src/utility/Sd2Card.cpp

```diff
@@ -541,20 +543,23 @@ uint8_t Sd2Card::setSpiClock(uint32_t clock)
 #endif
 //------------------------------------------------------------------------------
 // wait for card to go not busy
-uint8_t Sd2Card::waitNotBusy(uint16_t timeoutMillis) {
-  uint16_t t0 = millis();
+uint8_t Sd2Card::waitNotBusy(unsigned int timeoutMillis) {
+  unsigned int t0 = millis();
+  unsigned int d;
   do {
     if (spiRec() == 0XFF) return true;
+    d = millis() - t0;
   }
-  while (((uint16_t)millis() - t0) < timeoutMillis);
+  while (d < timeoutMillis);
   return false;
 }
 //------------------------------------------------------------------------------
 /** Wait for start block token */
 uint8_t Sd2Card::waitStartBlock(void) {
-  uint16_t t0 = millis();
+  unsigned int t0 = millis();
   while ((status_ = spiRec()) == 0XFF) {
-    if (((uint16_t)millis() - t0) > SD_READ_TIMEOUT) {
+    unsigned int d = millis() - t0;

```

Comment:
This is OK. Why not refactor the rest to use the same method for timeout detection.